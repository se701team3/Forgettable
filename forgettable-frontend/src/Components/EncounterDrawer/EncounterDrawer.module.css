@import '../../colours.css';

.EncounterDrawer {
    height: 100%;
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    box-sizing: border-box;
    padding: 0 50px;
    overflow-y: scroll;
    overflow-x: none;
}

.ContentContainer {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: flex-start;
}

.EncounterTitle {
    font-weight: 600;
    font-size: var(--text-xlarge);
    margin-top: 60px;
    margin-bottom: 0px;
}

.SubtitleContainer {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    /* margin: -10px 0 0 0; */
    margin: 20px 0;
}

.SubtitleContainer > * {
}

.EncounterSubtitle {
    font-weight: 600;
    font-size: var(--text-medium);
    white-space: nowrap;
    margin: 5px 0;
    padding-right: 15px;
}

.MiniPersonProfile {
    display: flex;
    flex-direction: row;
    align-items: center;
    padding-right: 20px;
    margin-top: 0;
    margin-bottom: 0;
    height: 40px
}

.MiniPersonProfile > p {
    font-size: var(--text-medium);
    font-weight: 400;
    margin-left: 10px;
}

.EncounterProperty {
    font-size: var(--text-small);
    margin: 5px 0;
    color: var(--txt2);
}

.ContentContainer > h2 {
    font-size: var(--text-medium);
    font-weight: 600;
    margin-bottom: 0;
    margin-top: 30px;
}

.EncounterDetails {
    font-size: var(--text-normal);
    color: var(--txt2);
}