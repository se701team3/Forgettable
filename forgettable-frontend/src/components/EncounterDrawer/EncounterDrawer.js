import React from 'react';
import classes from './EncounterDrawer.module.css';
import {Avatar, Drawer} from '@mui/material';
import PropTypes from 'prop-types';
import {
  getDateLastMetString,
  getFirstMetTimeString,
} from '../../functions/dateFormatter';
import {capitalise} from '../../functions/stringFormatter';
import convertSocialMediaNamesToIcons,
{convertSocialMediaToIcon} from '../../functions/socialMediaIconConverter';
import {IconButton} from '@mui/material';
import {getBirthdayString} from '../../functions/dateFormatter';
import CustomAvatarCollection from '../CustomAvatarCollection/CustomAvatarCollection';
import UnknownDetail from '../UnknownDetail/UnknownDetail';

/*
 * Side drawer for displaying information of an encounter.
 * The `open`, `encounterTitle`, and `encounterDetail` props
 * are required. Set `open` to true if the drawer should be always open.
 *
 * Author: Mercury Lin (lin8231)
 */
const EncounterDrawer = (props) => {
  return (
    <Drawer
      sx={{
        'width': '460px',
        'flexShrink': 0,
        '& .MuiDrawer-paper': {
          width: '460px',
          boxSizing: 'border-box',
        },
      }}
      variant="persistent"
      anchor="right"
      open={props.open}
      data-testid="drawer-element"
    >
      <div className={classes.EncounterDrawer}>
        <div className={classes.ContentContainer}>
          <h1 className={classes.EncounterTitle} data-testid={'title-element'}>
            {props.encounterTitle}
          </h1>
          <div className={classes.SubtitleContainer}>
            <h3 className={classes.EncounterSubtitle}>You encountered: </h3>
            <CustomAvatarCollection persons={props.persons}/>
          </div>
          <div
            className={classes.EncounterProperty}
            data-testid="date-met-element"
          >
            {'Date we met: '}
            {props.dateMet ?
            getDateLastMetString(props.dateMet) :
            <UnknownDetail/>}
          </div>
          <div
            className={classes.EncounterProperty}
            data-testid="location-element"
          >
            {'Location: '}
            {props.location ? props.location : <UnknownDetail/>}
          </div>
          <h2>Details:</h2>
          <p className={classes.EncounterDetails} data-testid="details-element">
            {props.encounterDetail}
          </p>
        </div>
      </div>
    </Drawer>
  );
};

EncounterDrawer.propTypes = {
  open: PropTypes.bool.isRequired,
  id: PropTypes.string,
  encounterTitle: PropTypes.string.isRequired,
  persons: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string,
    first_name: PropTypes.string.isRequired,
    last_name: PropTypes.string.isRequired,
    img: PropTypes.string,
  })).isRequired,
  dateMet: PropTypes.instanceOf(Date),
  location: PropTypes.string,
  encounterDetail: PropTypes.string.isRequired,
};

export default EncounterDrawer;
