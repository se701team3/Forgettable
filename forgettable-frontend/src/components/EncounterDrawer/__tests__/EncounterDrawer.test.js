import TestRenderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import {render, screen, fireEvent, cleanup} from '@testing-library/react';
import {
  within,
  getByText,
  getByTestId,
} from '@testing-library/dom';
import EncounterDrawer from '../EncounterDrawer';
import {BrowserRouter} from 'react-router-dom';

const PERSONS = [
  {

    id: '1',
    name: 'John Doe',
    img: 'https://www.w3schools.com/w3css/img_lights.jpg',
  },
  {

    id: '2',
    name: 'Mercury Lin',
    img: 'https://user-images.githubusercontent.com/62003343/158133392-b606b9d2-b05a-49dc-a4e8-9d6aed696d6b.gif',
  },
  {

    id: '3',
    name: 'Raina Song',
    img: 'https://cdn.pixabay.com/photo/2015/04/23/22/00/tree-736885__480.jpg',
  },
];

it('renders EncounterDrawer UI with correct hierarchy', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<EncounterDrawer
    open={true}
    dateMet={new Date('2020-01-01')}
    location="London"
    encounterTitle="Title"
    encounterDetails="We met on a train, it was a long journey"
    persons={PERSONS}
  />);
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});

afterEach(cleanup);

it('Title prop passed in is displayed correctly', () => {
  const title='Fun times';

  const {getByTestId} = render(
      <BrowserRouter>
        <EncounterDrawer
          open={true}
          persons={PERSONS}
          encounterTitle={title}
          encounterDetails="We met on a train, it was a long journey"
        /></BrowserRouter>);

  const {getByText} = within(getByTestId('title-element'));

  expect(getByText(title)).toBeInTheDocument();
});

it('displays the correct date met an existing date', () => {
  const dateMet = new Date('2000-01-01');

  const {getByTestId} = render(
      <BrowserRouter>
        <EncounterDrawer
          open={true}
          persons={PERSONS}
          dateMet={dateMet}
          encounterTitle="Title"
          encounterDetails="We met on a train, it was a long journey"
        /></BrowserRouter>);

  const {getByText} = within(getByTestId('date-met-element'));

  expect(
      getByText('Date we met: 01/01/2000'),
  ).toBeInTheDocument();
});


it('displays the correct location', () => {
  const location = 'London';

  const {getByTestId} = render(
      <BrowserRouter>
        <EncounterDrawer
          open={true}
          persons={PERSONS}
          location={location}
          encounterTitle="Title"
          encounterDetails="We met on a train, it was a long journey"
        /></BrowserRouter>);

  const {getByText} = within(getByTestId('location-element'));

  expect(
      getByText('Location: London'),
  ).toBeInTheDocument();
});


it('displays the details when the props is passed', () => {
  const details = 'We met on a train, it was a long journey';

  const {getByTestId} = render(
      <BrowserRouter>
        <EncounterDrawer
          open={true}
          persons={PERSONS}
          encounterDetails={details}
          encounterTitle="Title"
        /></BrowserRouter>);

  const {getByText} = within(getByTestId('location-element'));

  expect(
      getByText('Unknown'),
  ).toBeInTheDocument();
});

