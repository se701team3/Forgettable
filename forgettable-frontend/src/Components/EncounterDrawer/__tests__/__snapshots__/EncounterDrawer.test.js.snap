// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders EncounterDrawer UI with correct hierarchy 1`] = `
<ForwardRef(Drawer)
  anchor="right"
  data-testid="drawer-element"
  open={true}
  sx={
    Object {
      "& .MuiDrawer-paper": Object {
        "boxSizing": "border-box",
        "width": "460px",
      },
      "flexShrink": 0,
      "width": "460px",
    }
  }
  variant="persistent"
>
  <div
    className="EncounterDrawer"
  >
    <div
      className="ContentContainer"
    >
      <h1
        className="EncounterTitle"
        data-testid="title-element"
      />
      <div
        className="SubtitleContainer"
      >
        <h3
          className="EncounterSubtitle"
        >
          You encountered: 
        </h3>
        <div
          className="MiniPersonProfile"
        >
          <ForwardRef(Avatar)
            src="https://www.w3schools.com/w3css/img_lights.jpg"
            sx={
              Object {
                "height": "30px",
                "width": "30px",
              }
            }
          />
          <p
            data-testid="name-element"
          />
        </div>
        <div
          className="MiniPersonProfile"
        >
          <ForwardRef(Avatar)
            src="https://user-images.githubusercontent.com/62003343/158133392-b606b9d2-b05a-49dc-a4e8-9d6aed696d6b.gif"
            sx={
              Object {
                "height": "30px",
                "width": "30px",
              }
            }
          />
          <p
            data-testid="name-element"
          />
        </div>
        <div
          className="MiniPersonProfile"
        >
          <ForwardRef(Avatar)
            src="https://cdn.pixabay.com/photo/2015/04/23/22/00/tree-736885__480.jpg"
            sx={
              Object {
                "height": "30px",
                "width": "30px",
              }
            }
          />
          <p
            data-testid="name-element"
          />
        </div>
      </div>
      <div
        className="EncounterProperty"
        data-testid="date-met-element"
      >
        Date we met: 
        01/01/2020
      </div>
      <div
        className="EncounterProperty"
        data-testid="location-element"
      >
        Location: 
        London
      </div>
      <h2>
        Details:
      </h2>
      <p
        className="EncounterDetails"
        data-testid="details-element"
      />
    </div>
  </div>
</ForwardRef(Drawer)>
`;
